### PR TITLE
Travis build

### PR DESCRIPTION
**Pull request by hackaugusto**
Using a build environment variable for each kind of build, it might improve the build speed a bit. this will depend on the number of machines that travis has available for opensource projects.

~This is built on top of PR #220~

https://docs.travis-ci.com/user/speeding-up-the-build/#Parallelizing-your-builds-across-virtual-machines

### CODE REVIEW

**Comment by hackaugusto** [comment]
After experimenting a bit:
- The speed up for splitting is not much, but it's nice to have a faster build for unit tests / smart contracts and a way of restarting a single build
- The containers made a bigger win for build time

**Comment by LefterisJP** [comment]
Tests seem to still be failing with a polling timeoutl.

**Comment by hackaugusto** [comment]
Important note: This will require **manual** update of the solc and geth dependencies.

As a benefit we can run the build against multiple versions of these dependencies and we got a slight speed up from it (~30s for installing the ethereum package throught apt-get)

File: .travis.yml [hunk 1]
```diff
@@ -1,31 +1,53 @@
...
 python:
-- '2.7'
+  - '2.7'
+
+env:
+  global:
+    - GETH_URL='https://github.com/ethereum/go-ethereum/releases/download/v1.4.18/geth-linux-amd64-1.4.18-ef9265d0.tar.gz'
+    - GETH_VERSION='1.4.18'
+    - SOLC_URL='https://github.com/brainbot-com/solidity-static/releases/download/v0.4.2/solc'
+    - SOLC_VERSION='0.4.2'
```
Comment (LefterisJP) [inline .travis.yml]: It's gonna be quite annoying to have to specify new versions manually, but it makes sense.
Can you please update Solidity to `0.4.4` here? https://github.com/ethereum/solidity/releases

File: .travis/download_geth.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    cd geth*/
+    install -m 755 geth $HOME/.bin/geth-${GETH_VERSION}
+
+    success "geth ${GETH_VERSION} installed"
+else
+    info 'using cached geth'
+fi
+
+# always recreate the symlink since we dont know if it's poiting to a different
```
Comment (LefterisJP) [inline .travis/download_geth.sh]: poiting -> pointing

**Comment by LefterisJP** [comment]
I rebased your PR on top of master.

**Pull request merged**